samtranslator/model/apigateway.py

```diff
@@ -93,6 +93,38 @@ def make_auto_deployable(self, stage, swagger=None):
         stage.update_deployment_ref(self.logical_id)
 
 
+class ApiGatewayResponse(object):
+    def __init__(self, api_logical_id=None, response_parameters=None, response_templates=None, status_code=None):
+        self.api_logical_id = api_logical_id
+        self.response_parameters = response_parameters or {}
+        self.response_templates = response_templates or {}
+        self.status_code = status_code
+
+    def generate_swagger(self):
+        swagger = {
+            "responseParameters": self._add_prefixes(self.response_parameters),
+            "responseTemplates": self.response_templates
+        }
+
+        # Prevent "null" being written.
+        if self.status_code:
```

Comment:
Could you add some regex validation here?

Reply:
I've validated via:

```python
re.fullmatch(r'[0-9]{1,3}', str(status_code))
```

I hope that's what you meant.